### PR TITLE
chore: Better document use of StorageAPI

### PR DESCRIPTION
**Pull request by gibson042**
## Description

Consistently use and reference documentation regarding **StorageAPI** (typed as `KVStore`) and "enhanced storage API" (typed as `KVStorePlus`).

### Security Considerations

n/a

### Documentation Considerations

It would be nice to eliminate one concept from each pair, but I'm not taking that step in this PR.

### Testing Considerations

n/a

### CODE REVIEW

File: packages/SwingSet/src/lib/storageAPI.js [hunk 1]
```diff
@@ -19,10 +19,9 @@ export function insistStorageAPI(kvStore) {
 }
 
 /**
- * Assert function to ensure that something expected to be an enhanced storage
- * object actually implements the enhanced storage API.  It should be a storage
- * object that additionally has the methods `enumeratePrefixedKeys`,
- * `getPrefixedValues`, and `deletePrefixedKeys`.
+ * Assert function to ensure that an object implements the enhanced storage API.
+ * (StorageAPI plus methods { enumeratePrefixedKeys, getPrefixedValues,
+ * deletePrefixedKeys }), also known as "KVStorePlus".
```
Comment (gibson042) [inline packages/SwingSet/src/lib/storageAPI.js:24]: Given that the enhancements all relate to number-suffixed keys, should "enhanced storage API" be renamed to something like "numericalKeyStorageAPI" or "StorageAPIPlusNumericalKeys"?

File: packages/SwingSet/src/kernel/state/storageWrapper.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { assert } from '@agoric/assert';
 import { insistStorageAPI } from '../../lib/storageAPI.js';
 
-// We wrap a provided object implementing the Storage API (has/get/set/delete/getKeys)
-// and add some convenience methods.
+// We wrap a provided object implementing StorageAPI methods { has, getKeys,
```
Comment (warner) [inline packages/SwingSet/src/kernel/state/storageWrapper.js:4]: "host-provided object" would probably be the best way to describe it

File: packages/SwingSet/src/lib/storageAPI.js (hunk 1 above)
Comment (warner) [reply packages/SwingSet/src/lib/storageAPI.js:24]: Eh, I think that'd put too much detail into the name, it's not like we have a second enhancement layer that needs to be distinguished from this one.

**Review by warner — APPROVED** [review]
looks good

**Pull request merged**